qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/completions/QuteCompletionsForExpression.java

```diff
@@ -834,12 +847,39 @@ private void doCompleteExpressionForObjectPartWithParentNodes(Node part, Node no
 							}
 						}
 						break;
+					case WHEN:
+					case SWITCH:
+						// Completion for properties/methods of with object from #switch and #when
+						Parameter value = ((BaseWhenSection) section).getValueParameter();
+						if (value != null) {
+							String projectUri = template.getProjectUri();
+							ResolvedJavaTypeInfo whenSwitchJavaTypeInfo = javaCache.resolveJavaType(value, projectUri)
+									.getNow(null);
+							if (whenSwitchJavaTypeInfo != null
+									&& whenSwitchJavaTypeInfo.getJavaTypeKind() == JavaTypeKind.Enum) {

```

Comment:
To simplify the code 

`whenSwitchJavaTypeInfo.getJavaTypeKind() == JavaTypeKind.Enum`

I suggest that you create an isEnum() method in JavaTypeInfo

```java
public isEnum() {
    return getJavaTypeKind() == JavaTypeKind.Enum;
}
```

Do that for other type like Class, Interface, etc and don't miss to do that for JDT commons package too.